hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDWriteClient.java

```diff
@@ -482,4 +483,10 @@ protected void initWrapperFSMetrics() {
       HoodieWrapperFileSystem.setMetricsRegistry(registry, registryMeta);
     }
   }
+
+  @Override
+  protected void releaseResources() {
+    ((HoodieSparkEngineContext) context).getJavaSparkContext().getPersistentRDDs().values()
```

Comment:
@vinothchandar @minihippo This fix assumes that there is a single SparkHoodieWriteClient on the JVM. We have a usecase where a single process handles ingestion for multiple HUDI datasets each having their own SparkRDDWriteClient in different threads. The first write client which completes unpersists the RDDs for all other clients causing perf issues.

It may be better to have a write config (default false) for this fix.